Added new automatic job types. (#14785)
cockroachdb/docs: v22.1/show-jobs.md

Field | Description
------|------------
`job_id` | A unique ID to identify each job. This value is used if you want to control jobs (i.e., [pause](pause-job.html), [resume](resume-job.html), or [cancel](cancel-job.html) it).

> `job_type` | The type of job. Possible values: `SCHEMA CHANGE`, [`BACKUP`](backup.html), [`RESTORE`](restore.html), [`IMPORT`](import.html), and [`CREATE STATS`](create-statistics.html). <br><br> For job types for automatic jobs, see [Show automatic jobs](#show-automatic-jobs).

Reviewer: ```suggestion
`job_type` | The type of job. Possible values: `SCHEMA CHANGE`, [`BACKUP`](backup.html), [`RESTORE`](restore.html), [`IMPORT`](import.html), and [`CREATE STATS`](create-statistics.html). <br><br> For job types of automatic jobs, see [Show automatic jobs](#show-automatic-jobs).
```

~~~

~~~
    job_id           |       job_type                  |                    description                       |...
+--------------------+---------------------------------+------------------------------------------------------+...
  786475982730133505 | AUTO SPAN CONFIG RECONCILIATION | reconciling span configurations                      |...
  786483120403382274 | AUTO SQL STATS COMPACTION       | automatic SQL Stats compaction                       |...
  786476180299579393 | AUTO CREATE STATS               | Table statistics refresh for movr.public.promo_codes |...
...
(8 rows)
~~~

The job types of automatic jobs are:

> - `AUTO SPAN CONFIG RECONCILIATION`: a continuously running job that ensures that all declared [zone configurations](configure-zone.html) (`ALTER … CONFIGURE ZONE …`) are applied. For example, when `num_replicas = 7` is set on a table, the reconciliation job listens in on those changes and then informs the underlying [storage layer](architecture/storage-layer.html) to maintain 7 replicas for the table.

Reviewer: I think we normally capitalize after colons that introduce list items, including elsewhere on this page:

```suggestion
- `AUTO SPAN CONFIG RECONCILIATION`: A continuously running job that ensures that all declared [zone configurations](configure-zone.html) (`ALTER … CONFIGURE ZONE …`) are applied. For example, when `num_replicas = 7` is set on a table, the reconciliation job listens in on those changes and then informs the underlying [storage layer](architecture/storage-layer.html) to maintain 7 replicas for the table.
```

> - `AUTO SQL STATS COMPACTION`: an hourly job that truncates the internal `system.statement_statistics` and `system.transaction_statistics` table row count to the value in the `sql.stats.persisted_rows.max` [cluster setting](cluster-settings.html). Both tables contribute to the [`crdb_internal.statement_statistics`](crdb-internal.html#statement_statistics) and [`crdb_internal.transaction_statistics`](crdb-internal.html#transaction_statistics) tables respectively.

Reviewer: ```suggestion
- `AUTO SQL STATS COMPACTION`: An hourly job that truncates the internal `system.statement_statistics` and `system.transaction_statistics` table row counts to the value of the `sql.stats.persisted_rows.max` [cluster setting](cluster-settings.html). Both tables contribute to the [`crdb_internal.statement_statistics`](crdb-internal.html#statement_statistics) and [`crdb_internal.transaction_statistics`](crdb-internal.html#transaction_statistics) tables, respectively.
```

> - `AUTO CREATE STATS`: creates and updates [table statistics](cost-based-optimizer.html#table-statistics).

Reviewer: ```suggestion
- `AUTO CREATE STATS`: Creates and updates [table statistics](cost-based-optimizer.html#table-statistics).
```

~~~

~~~
        job_id       |         job_type          |                             description                             |                                         statement                                          | user_name |  status   | ...
  786483120403382274 | AUTO SQL STATS COMPACTION | automatic SQL Stats compaction                                      |                                                                                            | node      | succeeded | ...
  786476180299579393 | AUTO CREATE STATS         | Table statistics refresh for movr.public.promo_codes                | CREATE STATISTICS __auto__ FROM [110] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-30s' | root      | succeeded | ...
...
(7 rows)
~~~

### Show changefeed jobs

You can display specific fields relating to changefeed jobs by running `SHOW CHANGEFEED JOBS`. These fields include:

> * [`high_water_timestamp`](monitor-and-debug-changefeeds.html#monitor-a-changefeed): Guarantees all changes before or at this time have been emitted.

Reviewer: Not part of your changes, but we use `-` rather than `*` in lists and it wouldn't hurt to change these opportunistically.
